GetUrl.cs

```diff
@@ -134,29 +127,29 @@ private async Task<string> GetUrlFromBackend(string bcchCommand)
                 {
                     err = err.Replace("#< CLIXML\n", "");
                     var deserialized = PSSerializer.DeserializeAsList(err);
-                    err = String.Join("", deserialized);
-                    err = Regex.Replace(err, "\\u001b\\[[0-9]{1,3}m", "");
+                    err = string.Join("", deserialized);
+                    err = FixedRegex().Replace(err, "");
                     throw new Exception(err);
                 }
             }
             return await process.StandardOutput.ReadToEndAsync();
         }
 
-        private string validateTextParam(string paramName, string paramValue)
+        private static string ValidateTextParam(string paramName, string paramValue)
         {
             var validatedParam = "";
             if (!string.IsNullOrEmpty(paramValue))
                 validatedParam = $" -{paramName} \"{paramValue}\"";
             return validatedParam;
         }
 
-        private bool IsValidParamSet(string param)
+        private static bool IsValidParamSet(string param)
         {
             // Accept parameter without setting explict to true and with format "parameter=true"
-            return (param is not null && (param.All(char.IsWhiteSpace) || param.ToLower() == "true"));
+            return param is not null && (param.All(char.IsWhiteSpace) || param.Equals("true", StringComparison.CurrentCultureIgnoreCase));

```

Comment:
```suggestion
            return string.IsNullOrWhiteSpace(param) || param.Equals("true", StringComparison.CurrentCultureIgnoreCase);
```

With the new `<Nullable>enable</Nullable>` setting of the project (love this!), we can simplify this check.
The input param of this function should then also get a small change and should be allowed nullable `private static bool IsValidParamSet(string? param)`